src/validation.jl

```diff
@@ -160,10 +166,71 @@ function _validate(x, schema, ::Val{:not}, val, path::String)
 end
 
 # 9.2.2.1: if
-
+function _validate(x, schema, ::Val{:if}, val, path::String)
+    # ignore if without then or else
+    if haskey(schema, "then") || haskey(schema, "else")
+        ret = _if_then_else(x, schema, path)
+        return ret
+    end
+    return nothing
+end
 # 9.2.2.2: then
-
+function _validate(x, schema, ::Val{:then}, val, path::String)
+    # ignore then without if
+    if haskey(schema, "if")
+        ret = _if_then_else(x, schema, path)
+        return ret
+    end
+    return nothing
+end
 # 9.2.2.3: else
+function _validate(x, schema, ::Val{:else}, val, path::String)
+    # ignore else without if
+    if haskey(schema, "if")
+        ret = _if_then_else(x, schema, path)
+        return ret
+    end
+    return nothing
+end
+
+"""
+    _if_then_else(x, schema, path)
+
+The if, then and else keywords allow the application of a subschema based on the outcome of another schema. Details are in the link and the truth table is as follows:
+
+```
+┌─────┬──────┬──────┬────────┐
+│ if  │ then │ else │ result │       
+├─────┼──────┼──────┼────────┤ 
+│ T   │ T    │ n/a  │ T      │  
+│ T   │ F    │ n/a  │ F      │ 
+│ F   │ n/a  │ T    │ T      │ 
+│ F   │ n/a  │ F    │ F      │ 
+│ n/a │ n/a  │ n/a  │ T      │
+└─────┴──────┴──────┴────────┘
+https://json-schema.org/understanding-json-schema/reference/conditionals#ifthenelse
+```
+"""
+function _if_then_else(x, schema, path)
+    val_if = _validate(x, schema["if"], path)
+    val_then = if haskey(schema, "then")
+        _validate(x, schema["then"], path)
+    end
+    val_else = if haskey(schema, "else")
+        _validate(x, schema["else"], path)
+    end
+    if isnothing(val_if)
+        if isnothing(val_then) && isnothing(val_else)
+            return nothing
+        else
+            return val_then
+        end
+    end
+    if isa(val_if, SingleIssue)
+        return val_else
+    end
+    return nothing
+end
```

Comment:
I might have made a mistake, but I think you can write this code as:
```Julia
function _if_then_else(x, schema, path)
    if _validate(x, schema["if"], path) !== nothing
        if haskey(schema, "else")
            return _validate(x, schema["else"], path)
        end
    elseif haskey(schema, "then")
        return _validate(x, schema["then"], path)
    end
    return
end
```
but what about the case where you have a `if-then` and the `if` fails to validate?